src/olympia/devhub/tasks.py

```diff
@@ -208,7 +211,17 @@ def validate_upload(results, upload_pk, channel):
     Should only be called directly by Validator."""
     upload = FileUpload.objects.get(pk=upload_pk)
     data = validate_file_path(upload.path, channel)
-    return {**results, **json.loads(force_text(data))}
+    updated_results = {**results, **json.loads(force_text(data))}
+
+    # The tasks in the chord of the validation chain need `is_webextension` in
+    # `results['minimal_parsed_data']`. It often exists but it can be missing
+    # when `parse_data()` raised an error in Validator.
+    if not updated_results['minimal_parsed_data']:
+        linter_metadata = updated_results.get('metadata', {})
+        updated_results['minimal_parsed_data'] = {
+            'is_webextension': linter_metadata.get('is_webextension', False),
```

Comment:
I'm tempted to say we should directly use this all the time instead of updating results with `minimal_parsed_data`. Any reason not to ?

Reply:
Do you mean directly using `results['metadata']['is_webextension']` in the tasks?

Reply:
Yeah

Reply:
Ok, so I think I am going to update this patch with just that, but I will also revert https://github.com/mozilla/addons-server/issues/12436, which introduced an issue in `master`.